src/ChannelManager.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <unordered_map>
+#include <set>
+
+#include "StringUtils.h"
+
+
+class cClientHandle;
+class cCriticalSection;
+class cPluginLua;
+class cChannelCallback;
+
+
+// tolua_begin
+class cChannelManager
+{
+
+public:
+	typedef SharedPtr<cChannelCallback> cChannelCallbackRef;
+	typedef std::unordered_map<AString, cChannelCallbackRef> CallbackMap;
+
+	void AddClientToChannel      (cClientHandle & a_Client, const AString & a_Channel);
+	void AddClientToChannels     (cClientHandle & a_Client, const AString & a_Channels);
+	void RemoveClientFromChannel (cClientHandle & a_Client, const AString & a_Channel);
+	void RemoveClientFromChannels(cClientHandle & a_Client, const AString & a_Channels);
+	bool ClientHasChannel        (cClientHandle & a_Client, const AString & a_Channel);
+
+	void BroadcastChannelMessage (const AString & a_Channel,  const AString & a_Data, cClientHandle & a_Client, bool a_WorldOnly = true);
```

Comment:
Why does a "broadcast" function have a "client" param?

If it's only for the client's world, why not take a "world" param, defaulting to nullptr (-> all worlds)?